eth/client.go

```diff
@@ -79,13 +80,13 @@ func (c *PooledClient) inferError(err error) error {
 
 	switch {
 	case strings.Contains(err.Error(), "Cost of transaction exceeds sender balance"):
-		return concurrent.ErrCannotRecover{Cause: ErrExceedsBalance}
+		return errors.WithStack(concurrent.ErrCannotRecover{Cause: ErrExceedsBalance})
 	case strings.Contains(err.Error(), "Requested gas greater than block gas limit"):
-		return concurrent.ErrCannotRecover{Cause: ErrExceedsBlockLimit}
+		return errors.WithStack(concurrent.ErrCannotRecover{Cause: ErrExceedsBlockLimit})
 	case strings.Contains(err.Error(), "Invalid transaction nonce"):
-		return concurrent.ErrCannotRecover{Cause: ErrInvalidNonce}
+		return errors.WithStack(concurrent.ErrCannotRecover{Cause: ErrInvalidNonce})
 	default:
-		return err
+		return errors.WithStack(err)
```

Comment:
Aside: This looks janky as hell. We look at an error's human-readable message to determine its type?! I looks like these error strings come from `oasis-parity/ethcore/src/executed.rs`, and I guess that only gives us strings ... (Not a comment on your PR)

We should bubble up the whole `err` here, i.e. include `err` in the return value. Otherwise we're swallowing up valuable details of the error message.

Reply:
Good point, thanks! Since `owner.go` uses the predefined `ErrExceedsBalance`, `ErrExceedsBlockLimit`, and `ErrInvalidNonce`, I'll wrap these errors with the full message returned from `oasis-parity` so we retain the error message details.

This would lose the stack trace if the error object returned from `oasis-parity` includes a stackTrace (haven't checked yet). If we do need stack traces into oasis-parity in the future, I can revisit this. But the alternative to wrapping the `ErrExceedsBalance`, `ErrExceedsBlockLimit`, and `ErrInvalidNonce` errors is to do more string comparisons in `owner.go`, which I'd rather avoid if possible.